### ISSUE TITLE
no-construct-in-interface / no-construct-in-public-property-of-construct: eslint-plugin-awscdk misses a wrapper class that reuses the CDK class name (`class Bucket extends s3.Bucket`)

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#606]
### Describe the bug

In eslint-plugin-awscdk, `no-construct-in-interface` and `no-construct-in-public-property-of-construct` do not report a property typed with a wrapper class that reuses the CDK class name, e.g. `class Bucket extends s3.Bucket` (the usual "opinionated defaults" wrapper). A wrapper with a different name (`class MyBucket extends s3.Bucket`) is reported, and oxlint-plugin-awscdk reports both.

The hierarchy walk in `isResourceWithReadonlyInterface` keeps a "visited" set keyed by the class **name**. The user `Bucket` is visited first, so the CDK `Bucket` one level up is treated as already visited and the walk never reaches `BucketBase implements IBucket`.

`src/wrapper.ts`:

```ts
import * as s3 from "aws-cdk-lib/aws-s3";
import { Construct } from "constructs";

export class Bucket extends s3.Bucket {
  constructor(scope: Construct, id: string, props?: s3.BucketProps) {
    super(scope, id, { ...props, enforceSSL: true });
  }
}

export class MyBucket extends s3.Bucket {}

export interface WrapperProps {
  readonly wrapped: Bucket;
  readonly renamed: MyBucket;
}

export class Wrapper extends Construct {
  public readonly wrapped: Bucket;
  public readonly renamed: MyBucket;

  constructor(scope: Construct, id: string, props: WrapperProps) {
    super(scope, id);
    this.wrapped = props.wrapped;
    this.renamed = props.renamed;
  }
}
```

Rules enabled (`"error"`): `no-construct-in-interface`, `no-construct-in-public-property-of-construct`.

```
$ npx eslint src/wrapper.ts
  14:3  error  Interface property 'renamed' should not use CDK Construct type 'MyBucket'. Consider using an interface or type alias instead        awscdk/no-construct-in-interface
  19:3  error  Public property 'renamed' of Construct should not use Construct type 'MyBucket'. Consider using an interface or type alias instead  awscdk/no-construct-in-public-property-of-construct

$ npx oxlint --format=unix -c .oxlintrc.json src/wrapper.ts
src/wrapper.ts:13:3: Interface property 'wrapped' should not use CDK Construct type 'Bucket'. Consider using an interface or type alias instead. [Error/awscdk(no-construct-in-interface)]
src/wrapper.ts:14:3: Interface property 'renamed' should not use CDK Construct type 'MyBucket'. Consider using an interface or type alias instead. [Error/awscdk(no-construct-in-interface)]
src/wrapper.ts:18:3: Public property 'wrapped' of Construct should not use Construct type 'Bucket'. Consider using an interface or type alias instead. [Error/awscdk(no-construct-in-public-property-of-construct)]
src/wrapper.ts:19:3: Public property 'renamed' of Construct should not use Construct type 'MyBucket'. Consider using an interface or type alias instead. [Error/awscdk(no-construct-in-public-property-of-construct)]
```

### Expected Behavior

Lines 13 and 18 (`wrapped: Bucket`) are reported like lines 14 and 19: the wrapper extends `s3.Bucket -> BucketBase implements IBucket`, so `IBucket` is the type to expose.

### Current Behavior

eslint-plugin-awscdk reports only the differently named wrapper. Any class that shadows the CDK class name is invisible to both rules.

### Plugin

eslint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.2 (compared against oxlint-plugin-awscdk 1.0.2)

### Language Version

TypeScript 6.0.3

### Additional Context

```yaml
# machine-readable summary
rules: ["no-construct-in-interface", "no-construct-in-public-property-of-construct"]
symptom: ["false-negative", "cross-plugin-divergence"]
affected_packages: ["eslint-plugin-awscdk"]
root_cause: "packages/eslint/src/core/cdk-construct/type-checker/is-resource-with-readonly-interface.ts:49-57 (processedTypes is keyed by symbol.name, so a base class with the same name as the checked class is skipped as already visited)"
verified_on: "main @ 9caea55 (eslint-plugin-awscdk 4.4.2, oxlint-plugin-awscdk 1.0.2)"
related_issues: ["#572", "#602"]
```

Base chain of the user `Bucket`, from the TypeScript compiler API (`implements` per level):

```
Bucket (src/wrapper.ts)                     implements=[]
  Bucket (aws-cdk-lib/aws-s3/lib/bucket.d.ts) implements=[]            <- skipped: name already in processedTypes
    BucketBase (aws-s3/lib/bucket.d.ts)       implements=[IBucket,IEncryptedResource]
      Resource (core/lib/resource.d.ts)       implements=[IResource]
        Construct (constructs)                implements=[IConstruct]
```

Verified fix direction: keying `processedTypes` by the `Symbol` object instead of `symbol.name` makes eslint report all four lines, with `examples/` still at 43 errors. The oxlint finder has the same name-keyed set but is not affected today because corsa's `getImplementedTypesOfType()` already returns the inherited `IBucket` at the user class level; keying by identity there as well keeps the two mirrored.

Reproduction environment:

- macOS 14.1, Node.js v24.18.0, pnpm 10.29.3
- aws-cdk-lib 2.268.0 / constructs 10.8.1
- ESLint 10.9.1 + typescript-eslint 8.69.0 + TypeScript 6.0.3
- oxlint 1.81.0 + oxlint-tsgolint 7.0.2001 (corsa-oxlint 1.13.1)
